src/pubsub/src/subscriber/builder.rs

```diff
@@ -0,0 +1,174 @@
+        self.ack_deadline_seconds = v.into();
+        self
+    }
+
+    /// Flow control settings for the maximum number of outstanding messages.
+    ///
+    /// The server will stop sending messages to a client when this many
+    /// messages are outstanding (i.e. that have not been acked or nacked).
+    ///
+    /// The server resumes sending messages when the outstanding message count
+    /// drops below this value.
+    ///
+    /// Use a value <= 0 to set no limit on the number of outstanding messages.
+    ///
+    /// # Example
+    ///
+    /// ```no_rust
+    /// # use google_cloud_pubsub::client::Subscriber;
+    /// # async fn sample() -> anyhow::Result<()> {
+    /// # let client = Subscriber::builder().build().await?;
+    /// let session = client.streaming_pull("projects/my-project/subscriptions/my-subscription")
+    ///     .set_max_outstanding_messages(100000)
+    ///     .start();
+    /// # Ok(()) }
+    /// ```
+    pub fn set_max_outstanding_messages<T: Into<i64>>(mut self, v: T) -> Self {
+        self.max_outstanding_messages = v.into();
+        self
+    }
+
+    /// Flow control settings for the maximum number of outstanding bytes.
+    ///
+    /// The server will stop sending messages to a client when this many bytes
+    /// of messages are outstanding (i.e. that have not been acked or nacked).
+    ///
+    /// The server resumes sending messages when the outstanding byte count
+    /// drops below this value.
+    ///
+    /// Use a value <= 0 to set no limit on the number of outstanding bytes.
+    ///
+    /// # Example
+    ///
+    /// ```no_rust
+    /// # use google_cloud_pubsub::client::Subscriber;
+    /// # async fn sample() -> anyhow::Result<()> {
+    /// # let client = Subscriber::builder().build().await?;
+    /// let session = client.streaming_pull("projects/my-project/subscriptions/my-subscription")
+    ///     .set_max_outstanding_bytes(1024 * 1024 * 1024)
+    ///     .start();
+    /// # Ok(()) }
+    /// ```
+    pub fn set_max_outstanding_bytes<T: Into<i64>>(mut self, v: T) -> Self {
+        self.max_outstanding_bytes = v.into();
+        self
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::super::stub::tests::MockStub;
+    use super::*;
+
+    #[test]
+    fn reasonable_defaults() {
+        let mock = MockStub::new();
+        let builder = StreamingPull::new(
+            Arc::new(mock),
+            "projects/my-project/subscriptions/my-subscription".to_string(),
+        );
+        assert_eq!(
+            builder.subscription,
+            "projects/my-project/subscriptions/my-subscription"
+        );
+        assert_eq!(builder.ack_deadline_seconds, 10);
+        assert!(
+            1000000 > builder.max_outstanding_messages && builder.max_outstanding_messages > 100,
+            "{}",
+            builder.max_outstanding_messages
+        );
+        assert!(
+            builder.max_outstanding_bytes > 100000,
```

Comment:
nit: Could make sense to use a number of KiB instead, since we are mostly dealing with bytes in terms of increments of 1024 (maybe use 100 KiB = 100 * 1024)?